crates/payload/builder/src/payload.rs

```diff
@@ -177,9 +177,24 @@ impl PayloadBuilderAttributes {
         // configure evm env based on parent block
         let mut cfg = CfgEnv::default();
         cfg.chain_id = chain_spec.chain().id();
+
         // ensure we're not missing any timestamp based hardforks
         cfg.spec_id = revm_spec_by_timestamp_after_merge(chain_spec, self.timestamp);
 
+        // if the parent block did not have excess blob gas (i.e. it was pre-cancun), but it is
+        // cancun now, we need to set the excess blob gas to the default value
+        let excess_blob_gas = parent.next_block_blob_fee().map_or_else(
+            || {
+                if cfg.spec_id == SpecId::CANCUN {
```

Comment:
```suggestion
                if cfg.spec_id >= SpecId::CANCUN {
```

Reply:
I think >= would cover all cases were additional forks are activated at block 0, but probably not a realistic edge case so not sure if even useful to check

Reply:
in this case the genesis block should have it as 0 so we don't need to check this